server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java

```diff
@@ -299,4 +301,42 @@ void givenTwoTaskManagerAndATaskRanPerTaskManagerListingThemOnEachShouldShowBoth
                 });
         }
     }
+
+    @Test
+    void givenTwoTaskManagerIfTheFirstOneIsDownTheSecondOneShouldBeAbleToRunTheRemainingTasks(CountDownLatch countDownLatch) throws InterruptedException {
+        try (EventSourcingTaskManager taskManager1 = taskManager()) {
+            try (EventSourcingTaskManager taskManager2 = taskManager(HOSTNAME_2)) {
+                ImmutableBiMap<TaskManager, Hostname> hostNameByTaskManager = ImmutableBiMap.of(taskManager1, HOSTNAME, taskManager2, HOSTNAME_2);
+                TaskId firstTask = taskManager1.submit(new MemoryReferenceTask(() -> {
+                    countDownLatch.await();
+                    return Task.Result.COMPLETED;
+                }));
+
+                awaitUntilTaskHasStatus(firstTask, TaskManager.Status.IN_PROGRESS, taskManager1);
+
+                Hostname nodeRunningFirstTask = taskManager1.getExecutionDetails(firstTask).getRanNode().get();
+                Hostname otherNode = getOtherNode(hostNameByTaskManager, nodeRunningFirstTask);
+                EventSourcingTaskManager taskManagerRunningFirstTask = (EventSourcingTaskManager) hostNameByTaskManager.inverse().get(nodeRunningFirstTask);
```

Comment:
why would you want to cast here? Can't you define the collection with the right type?